monitoring/monitoring.go

```diff
@@ -26,10 +33,33 @@ func Handler() http.Handler {
 	return promhttp.HandlerFor(longhornCustomRegistry, promhttp.HandlerOpts{})
 }
 
-func InitMonitoringSystem(logger logrus.FieldLogger, currentNodeID string, ds *datastore.DataStore) {
+func InitMonitoringSystem(logger logrus.FieldLogger, currentNodeID string, ds *datastore.DataStore, kubeconfigPath string) error {
+	namespace := os.Getenv(types.EnvPodNamespace)
+	if namespace == "" {
+		logger.Warnf("Cannot detect pod namespace, environment variable %v is missing, "+
+			"using default namespace", types.EnvPodNamespace)
+		namespace = corev1.NamespaceDefault
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		return errors.Wrap(err, "unable to get client config. Failed to init the monitoring system")
+	}
+
+	kubeMetricsClient, err := metricsclientset.NewForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, "unable to get k8s metrics client. Failed to init the monitoring system")
```

Comment:
Do we want to fail all metrics collection in this case or just proceed with the setup then, log an error in the 
`InstanceManagerCollector` as well as the `ManagerCollector`?

The constructor can then just have an if kubeMetricsClient is nil -> log
and the collect method can just have an  if kubeMetricsClient is nil -> do nothing

If we are saying that metrics are mandatory, then failing here is okay otherwise if metrics are best effort then we should proceed with regular longhorn functionality.

Reply:
Great point! I agree that all metrics should be best-effort. Added the logic to skip instantiating InstanceManagerCollector and ManagerCollector  when failing to create kubeMetricsClient.